PYME/experimental/_triangle_mesh.pyx

```diff
@@ -1877,20 +1876,19 @@ cdef class TriangleMesh(TrianglesBase):
                 continue
             
             _twin = curr_edge.twin
-            twin_edge = &self._chalfedges[_twin]
 
+            v2, v4 = 0, 0
             target_valence = VALENCE
             if _twin == -1:
                 # boundary
                 target_valence = BOUNDARY_VALENCE
+            else:
+                twin_edge = &self._chalfedges[_twin]
+                v2 = self._cvertices[twin_edge.vertex].valence - target_valence
+                v4 = self._cvertices[self._chalfedges[twin_edge.next].vertex].valence - target_valence
 
-            # Pre-flip vertices
```

Comment:
Think it would be useful to still comment somewhere that v1 and v2 are the pre-flip valences and v3, v4 the post flip ones